web/pdf_viewer.js

```diff
@@ -1099,10 +1114,27 @@ class PDFViewer {
       newScale * PixelsPerInch.PDF_TO_CSS_UNITS
     );
 
-    const updateArgs = { scale: newScale };
+    const mustPostponeDrawing = drawingDelay >= 0 && drawingDelay < 1000;
+    const updateArgs = {
+      scale: newScale,
+    };
+    if (mustPostponeDrawing) {
+      updateArgs.postponeDrawingDelay = drawingDelay;
+    }
     for (const pageView of this._pages) {
       pageView.update(updateArgs);
     }
+
+    if (this.#scaleTimeoutId !== null) {
+      clearTimeout(this.#scaleTimeoutId);
+    }
```

Comment:
Do we perhaps want to try and reset this timeout in more places as well, I'm thinking in particular about the `PDFViewer.update` method?

Consider the following, somewhat contrived, series of events:
 1. The user modifies the new preference to its maximum value.
 2. The user zoom in/out quickly a few times, using either a mouse or keyboard shortcuts.
 3. The user then *immediately* rotates the document.

In this case we immediately re-render the rotated document, however the `#scaleTimeoutId` may still be running and it could thus force us to almost immediately invalidate/re-render again.

Note here that in this case, since the scale is a numerical value and because of the `if (this.#isSameScale(newScale)) { ... }`-branch above, we'd not actually reset the timeout in this case.